itest/rfq_test.go

```diff
@@ -0,0 +1,353 @@
+	// expected scid. Bob will then use the scid to identify the HTLC as
+	// relating to the accepted quote.
+	acceptedQuote := acceptedQuotes.AcceptedQuotes[0]
+	t.Logf("Accepted quote scid: %d", acceptedQuote.Scid)
+	scid := lnwire.NewShortChanIDFromInt(acceptedQuote.Scid)
+
+	// Use the agreed upon scid found in the accepted quote to construct a
+	// route hop hint for the Alice->Bob step of the payment. The route hop
+	// hint will be included in the invoice that Carol hands to Alice.
+	aliceBobHopHint := &lnrpc.HopHint{
+		NodeId: ts.BobLnd.PubKeyStr,
+		ChanId: scid.ToUint64(),
+		FeeBaseMsat: uint32(
+			chainreg.DefaultBitcoinBaseFeeMSat,
+		),
+		FeeProportionalMillionths: uint32(
+			chainreg.DefaultBitcoinFeeRate,
+		),
+		CltvExpiryDelta: chainreg.DefaultBitcoinTimeLockDelta,
+	}
+	routeHints := []*lnrpc.RouteHint{
+		{
+			HopHints: []*lnrpc.HopHint{
+				aliceBobHopHint,
+			},
+		},
+	}
+
+	// Carol can now finalise the invoice and hand it over to Alice for
+	// settlement.
+	addInvoiceResp := ts.CarolLnd.RPC.AddInvoice(&lnrpc.Invoice{
+		ValueMsat:  int64(bidAmt),
+		RouteHints: routeHints,
+	})
+	invoice := ts.CarolLnd.RPC.LookupInvoice(addInvoiceResp.RHash)
+
+	// Register to receive RFQ events from Bob's tapd node. We'll use this
+	// to wait for Bob to receive the HTLC with the asset transfer specific
+	// scid.
+	bobEventNtfns, err := ts.BobTapd.SubscribeRfqEventNtfns(
+		ctxb, &rfqrpc.SubscribeRfqEventNtfnsRequest{},
+	)
+	require.NoError(t.t, err)
+
+	// Alice pays the invoice.
+	t.Log("Alice paying invoice")
+	req := &routerrpc.SendPaymentRequest{
+		PaymentRequest: invoice.PaymentRequest,
+		TimeoutSeconds: int32(wait.PaymentTimeout.Seconds()),
+		FeeLimitMsat:   math.MaxInt64,
+	}
+	ts.AliceLnd.RPC.SendPayment(req)
+	t.Log("Alice payment sent")
+
+	// At this point Bob should have received a HTLC with the asset transfer
+	// specific scid. We'll wait for Bob to publish an accept HTLC event and
+	// then validate it against the accepted quote.
+	waitErr = wait.NoError(func() error {
+		t.Log("Waiting for Bob to receive HTLC")
+
+		event, err := bobEventNtfns.Recv()
+		require.NoError(t.t, err)
+
+		acceptHtlc, ok := event.Event.(*rfqrpc.RfqEvent_AcceptHtlc)
+		if ok {
+			require.Equal(
+				t.t, acceptedQuote.Scid,
+				acceptHtlc.AcceptHtlc.Scid,
+			)
+			t.Log("Bob has accepted the HTLC")
+			return nil
+		}
+
+		return fmt.Errorf("unexpected event: %v", event)
+	}, defaultWaitTimeout)
+	require.NoError(t.t, waitErr)
+
+	// Close event streams.
+	err = carolEventNtfns.CloseSend()
+	require.NoError(t.t, err)
+
+	err = bobEventNtfns.CloseSend()
+	require.NoError(t.t, err)
+}
+
+// newLndNode creates a new lnd node with the given name and funds its wallet
+// with the specified outputs.
+func newLndNode(name string, outputFunds []btcutil.Amount,
+	ht *lntest.HarnessTest) *node.HarnessNode {
+
+	lndArgs := []string{
+		"--default-remote-max-htlcs=483",
+		"--dust-threshold=5000000",
```

Comment:
Why is this necessary?

Reply:
Ah I was just matching the standby nodes to get this to work. I'll try to remove this now.